foladipo/news-flash-cp1
# Implement CP1 defense feedback

#### What does this PR do?
Implements the feedback received during my checkpoint 1 defense. These include removing the use of `ejs` and converting this project to a single page app (SPA) instead, untracking built files (like `bundle.js`) and hosting images with a CDN, testing all Actions, extracting all keys and tokens to a `.env` file and reading them with the `dotenv` npm package, adding search functionality for news sources, retrieving the list of sources and their available sort options when the dashboard is opened, installing `eslint` and fixing the errors it flags etc

#### Description of task to be completed?
Implement feedback given during project defense.

#### How should this be manually tested?
- In a terminal/command prompt on your computer, switch to the `feature-chore/HEupl2vT/implement-defense-feedback` branch of this repo and run the project.
- Also, run `npm test` in the terminal to see all the tests.
- Check the README of this project to see the update to test coverage and GPA badges.
- Navigate the different dirrectories to see their new or renamed or rearranged contents.

#### Any background context you want to provide?
N/A

#### What are the relevant Trello cards?
https://trello.com/c/HEupl2vT

#### Screenshots


<img width="1028" alt="screen shot 2017-05-06 at 3 44 07 pm" src="https://cloud.githubusercontent.com/assets/26963575/25773274/e5882fb6-3272-11e7-95e3-e8ba091c9d86.png">
